app/src/androidTest/java/org/mozilla/fenix/ui/robots/SearchRobot.kt

```diff
@@ -110,31 +111,34 @@ class SearchRobot {
         }
     }
 
-    fun verifyFirefoxSuggestResults(rule: ComposeTestRule, searchSuggestion: String) {
+    fun verifyFirefoxSuggestResults(rule: ComposeTestRule, searchTerm: String, vararg searchSuggestions: String) {
         rule.waitForIdle()
         for (i in 1..RETRY_COUNT) {
             try {
-                rule.onNodeWithTag("mozac.awesomebar.suggestions")
-                    .performScrollToNode(hasText(searchSuggestion))
-                    .assertExists()
-                break
+                for (searchSuggestion in searchSuggestions) {
+                    mDevice.waitForObjects(mDevice.findObject(UiSelector().textContains(searchSuggestion)))
+                    rule.onNodeWithTag("mozac.awesomebar.suggestions")
+                        .performScrollToNode(hasText(searchSuggestion))
+                        .assertExists()
+                }
```

Comment:
Don't we still need the `break` to stop the iteration for `RETRY_COUNT`?

Reply:
Yes, added the `break`.